src/daemon.py

```diff
@@ -199,6 +203,19 @@ def write_coingecko_prices(
         logger.error("Error inserting record: %s", e)
 
 
+def get_price(
+    PriceProviders: List, block_number: int, token_address: str
+) -> Optional[Tuple[float, str]]:
+    for provider in PriceProviders:
+        try:
+            price = provider.get_price(block_number, token_address)
+            if price is not None:
+                return price, provider.__class__.__name__
+        except Exception as e:
+            logger.error("Error getting price from provider: %s", e)
```

Comment:
Judging by how common we expect this to be, maybe an info message would be more appropriate compared to the error emitted here.